File: chatGPT_example_java/src/main/java/com/chatgpt/example/config/WebsocketConfig.java
package com.chatgpt.example.config;

import com.chatgpt.example.service.StreamCompletionHandler;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
@RequiredArgsConstructor
public class WebsocketConfig implements WebSocketConfigurer {

  private final StreamCompletionHandler streamCompletionHandler;

  @Override
  public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
    registry.addHandler(streamCompletionHandler, "/chat/stream").setAllowedOrigins("*");
  }
}

File: chatGPT_example_kotlin/src/main/kotlin/com/example/chatgpt_example_kotlin/common/config/WebsocketConfig.kt
package com.example.chatgpt_example_kotlin.common.config

import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry

@Configuration
@EnableWebSocket
class WebsocketConfig: WebSocketConfigurer {

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
    }
}